src/main/java/com/github/javabdd/BDDFactory.java

```diff
@@ -1327,6 +1344,104 @@ public CacheStats getCacheStats() {
         return cachestats;
     }
     
+    /**
+     * Stores statistics about the maximum memory usage.
+     * 
+     * @author mgoorden
+     */
+    public static class MaxMemoryStats {
+    	public boolean measuring = false;
+        public long maxUsedMemory;
+        public int maxUsedBddNodes;
+        
+        protected MaxMemoryStats() { }
+        
+        public void enableMeasurments() {
+        	measuring = true;
+        }
+        
+        public void newMeasurement(int newBddNodes) {
+            maxUsedBddNodes = Math.max(newBddNodes, maxUsedBddNodes);
+        }
+        
+        public void newMeasurement(int newBddNodes, long newMemory) {
+            maxUsedBddNodes = Math.max(newBddNodes, maxUsedBddNodes);
+            maxUsedMemory = Math.max(newMemory, maxUsedMemory);
+        }
+        
+        public int getMaxNodes() {
+        	return maxUsedBddNodes;
+        }
+        
+        // 'max memory usages'.
+        // Measuring max memory usages fluctuates over time, tool implementation,
+        // used hardware, etc. Therefore, if one wants to collect this information,
+        // he/she should uncomment lines below and in JFactory.java.
+        //public long getMaxUsedMemory() {
+        //	return maxUsedMemory;
+        //}
+        // End of 'max memory usages' comment.
+    }
+    
+    /**
+     * Singleton object for maximum memory statistics.
+     */
+    protected MaxMemoryStats maxmemorystats = new MaxMemoryStats();
+
+    /**
+     * <p>Return the current maximum memory statistics for this BDD factory.</p>
+     *
+     * @return  maximum memory statistics
+     */
+    public MaxMemoryStats getMemoryStats() {
+        return maxmemorystats;
+    }
+    
+    /**
+     * Stores continuously statistics about the memory usage and BDD operations.
+     * 
+     * @author mgoorden
+     */
+    public static class ContinousStats {
+    	public boolean measuring = false;
+        public List<Integer> contNodes = new ArrayList<Integer>();
```

Comment:
I don't like the max in your suggestion, as it might suggest some max operator being applied. This array stores the number of Bdd nodes over time. I opted to go for `contUsedBddNodes`.